src/cli/commands/match.ts

```diff
@@ -0,0 +1,44 @@
+import { contractDeployInfo } from '../../contract';
+import { getSaddle } from '../../saddle';
+
+import { info, debug, warn, error } from '../logger';
+import { describeProvider } from '../../utils';
+
+import { diffStringsUnified } from 'jest-diff';
+
+export async function match(network: string, address: string, contractName: string, contractArgs: any[], trace: boolean, verbose: number): Promise<void> {
+  let saddle = await getSaddle(network);
+
+  info(`Matching contract at ${address} to ${contractName} with args ${JSON.stringify(contractArgs)}`, verbose);
+
+  const callOptions = {
+    ...saddle.network_config.defaultOptions,
+    from: saddle.account
+  };
+
+  let contractCode = await saddle.web3.eth.getCode(address);
+
+  if (contractCode === '0x') {
+    throw new Error(`Match Failed: No contract code found at ${address} on ${network}`);
+  }
+
+  let deploymentABI = await contractDeployInfo(saddle.web3, network, contractName, contractArgs, saddle.network_config, saddle.network_config.defaultOptions, callOptions);
+
+  let expectedBytecode = await saddle.web3.eth.call({
+    ...callOptions,
+    to: undefined,
+    data: deploymentABI
+  });
+
+  if (expectedBytecode === '0x') {
+    throw new Error(`Match Failed: Contract "${contractName}" creation reverted on ${network}`);
+  }
+
+  if (expectedBytecode != contractCode) {
+    error(diffStringsUnified(expectedBytecode, contractCode), verbose);
+
+    throw new Error(`Match Failed: Mismatched bytecode`);
+  }
+
+  info(`✅ Successfully matched ${contractName} to ${address} with args ${JSON.stringify(contractArgs)}`, verbose);
```

Comment:
wondering if we will run into same issues as etherscan verifying, at least we can debug though!

Reply:
It's been *super finicky* in my testing so far, honestly. If the code matches directly, it passes every time, but if I add another random contract that I don't even reference, it fails. It's almost like there's a checksum of the AST or something there. That said, I use jest-diff which pretty-prints the exact mismatch, so it makes it pretty easy to debug and spot and we can do it locally.

Reply:
Feeling some sympathy for etherscan now?